tests/gold_tests/pluginTest/certifier/certifier.test.py

```diff
@@ -0,0 +1,170 @@
+            "proxy.config.ssl.server.cert.path": f'{self.ts.Variables.SSLDir}',
+            "proxy.config.ssl.server.private_key.path": f'{self.ts.Variables.SSLDir}',
+        })
+        self.ts.Disk.ssl_multicert_config.AddLine(
+            'dest_ip=* ssl_cert_name=server.pem ssl_key_name=server.key'
+        )
+        self.ts.Disk.remap_config.AddLine(
+            f"map / http://127.0.0.1:{self.server.Variables.http_port}/",
+        )
+        self.ts.Disk.plugin_config.AddLine(
+            f'certifier.so -s {os.path.join(self.certPathDest, "store")} -m 1000 -c {os.path.join(self.certPathDest, "ca.cert")} -k {os.path.join(self.certPathDest, "ca.key")} -r {os.path.join(self.certPathDest, "ca-serial.txt")}')
+        # Verify logs for dynamic generation of certs
+        self.ts.Disk.traffic_out.Content += Testers.ContainsExpression(
+            "Creating shadow certs",
+            "Verify the certifier plugin generates the certificate dynamically.")
+
+    def runHTTPSTraffic(self):
+        tr = Test.AddTestRun("Test dynamic generation of certs")
+        tr.AddVerifierClientProcess(
+            "client1",
+            self.httpsReplayFile,
+            http_ports=[self.ts.Variables.port],
+            https_ports=[self.ts.Variables.ssl_port],
+            other_args='--thread-limit 1')
+        tr.Processes.Default.StartBefore(self.server)
+        tr.Processes.Default.StartBefore(self.ts)
+        tr.StillRunningAfter = self.server
+        tr.StillRunningAfter = self.ts
+
+    def verifyCert(self, certPath):
+        tr = Test.AddTestRun("Verify the content of the generated cert")
+        tr.Processes.Default.Command = f'openssl x509 -in {certPath} -text -noout'
+        tr.Processes.Default.ReturnCode = 0
+        # Verifiy certificate content
+        tr.Processes.Default.Streams.All += Testers.ContainsExpression(
+            "Subject: CN = www.tls.com", "Subject should match the host in the request")
+        tr.Processes.Default.Streams.All += Testers.ContainsExpression(
+            r"X509v3 extensions:\n.*X509v3 Subject Alternative Name:.*\n.*DNS:www.tls.com",
+            "Should contain the SAN extension",
+            reflags=re.MULTILINE)
+
+    def run(self):
+        # the certifier plugin generates the cert and store it in a directory
+        # named with the first three character of the md5 hash of the hostname
+        genCertDirPath = os.path.join(self.certPathDest,
+                                      'store',
+                                      str(hashlib.md5(self.host.encode('utf-8')).hexdigest()[:3]))
+        genCertPath = os.path.join(genCertDirPath, self.host + ".crt")
+        genCertDir = Test.Disk.Directory(genCertDirPath)
+        genCertFile = Test.Disk.File(genCertPath)
+        genCertDir.Exists = False
+        genCertFile.Exists = False
+        self.runHTTPSTraffic()
+        genCertDir.Exists = True
+        genCertFile.Exists = True
```

Comment:
I believe the latter Exists calls are simply overriding the former Exists ones. This can be a bit confusing, but AuTest is declarative rather than procedural in nature. Thus `genCertDir.Exists = True` isn't saying, "At this point in the code, check that the file exists. Rather it is saying, "At the end of the test run, check that `genCertDir` exists.